packages/quantic/force-app/main/default/lwc/quanticFeedback/__tests__/quanticFeedback.test.js

```diff
@@ -0,0 +1,326 @@
+          state: 'liked',
+          hideExplainWhyButton: true,
+        });
+        await flushPromises();
+
+        const successMessage = element.shadowRoot.querySelector(
+          selectors.successMessage
+        );
+        const explainWhyButton = element.shadowRoot.querySelector(
+          selectors.explainWhyButton
+        );
+
+        expect(successMessage).not.toBeNull();
+        expect(successMessage.textContent).toBe(exampleSuccessMessage);
+        expect(explainWhyButton).toBeNull();
+      });
+    });
+  });
+
+  describe('when the component is in a disliked state', () => {
+    it('should display the feedback buttons in a disliked state', async () => {
+      const element = createTestComponent({
+        ...defaultOptions,
+        state: 'disliked',
+      });
+      await flushPromises();
+
+      const likeButton = element.shadowRoot.querySelector(selectors.likeButton);
+      const likeButtonIcon = element.shadowRoot.querySelector(
+        selectors.likeButtonIcon
+      );
+      const dislikeButton = element.shadowRoot.querySelector(
+        selectors.dislikeButton
+      );
+      const dislikeButtonIcon = element.shadowRoot.querySelector(
+        selectors.dislikeButtonIcon
+      );
+      expect(likeButtonIcon.variant).toBeNull();
+      expect(dislikeButtonIcon.variant).toBe('error');
+      expect(likeButton.classList.contains('feedback__button--neutral')).toBe(
+        true
+      );
+      expect(
+        dislikeButton.classList.contains('feedback__button--disliked')
+      ).toBe(true);
+    });
+
+    it('should display the success message', async () => {
+      const element = createTestComponent({
+        ...defaultOptions,
+        state: 'disliked',
+      });
+      await flushPromises();
+
+      const successMessage = element.shadowRoot.querySelector(
+        selectors.successMessage
+      );
+      const explainWhyButton = element.shadowRoot.querySelector(
+        selectors.explainWhyButton
+      );
+
+      expect(successMessage).not.toBeNull();
+      expect(successMessage.textContent).toBe(exampleSuccessMessage);
+      expect(explainWhyButton).not.toBeNull();
+    });
+
+    describe('when the explain why button is clicked', () => {
+      it('should dispatch the "pressexplainwhy" event', async () => {
+        const element = createTestComponent({
+          ...defaultOptions,
+          state: 'disliked',
+        });
+        setupEventListeners(element);
+        await flushPromises();
+
+        const explainWhyButton = element.shadowRoot.querySelector(
+          selectors.explainWhyButton
+        );
+        explainWhyButton.click();
+
+        expect(explainWhyButton).not.toBeNull();
+        expect(functionsMocks.exampleHandleExplainWhy).toHaveBeenCalledTimes(1);
+      });
+    });
+
+    describe('when the hideExplainWhy property is set to true', () => {
+      it('should display the success message without the explain why button', async () => {
```

Comment:
This I understand we want to remove the explain why when the feedback is submitted correct?

Reply:
In the Smart Snippet feature for example, we want to display the `explain why` button only when the user gives a negative feedback. that's why the property `hideExplainWhy` has been added, to give us easy control on the display of the `explain why` button.